jni/src/org_opensearch_knn_jni_NmslibService.cpp

```diff
@@ -12,79 +12,102 @@
 }
 
-JNIEXPORT void JNICALL Java_org_opensearch_knn_jni_NmslibService_createIndex(JNIEnv * env, jclass cls, jintArray idsJ,
-                                                                             jlong vectorsAddressJ, jint dimJ, jstring indexPathJ,
-                                                                             jobject parametersJ)
-{
-    try {
-        knn_jni::nmslib_wrapper::CreateIndex(&jniUtil, env, idsJ, vectorsAddressJ, dimJ, indexPathJ, parametersJ);
-    } catch (...) {
-        jniUtil.CatchCppExceptionAndThrowJava(env);
-    }
+JNIEXPORT void JNICALL Java_org_opensearch_knn_jni_NmslibService_createIndex(JNIEnv *env,
+                                                                             jclass cls,
+                                                                             jintArray idsJ,
+                                                                             jlong vectorsAddressJ,
+                                                                             jint dimJ,
+                                                                             jstring indexPathJ,
+                                                                             jobject parametersJ) {
+  try {
+    knn_jni::nmslib_wrapper::CreateIndex(&jniUtil, env, idsJ, vectorsAddressJ, dimJ, indexPathJ, parametersJ);
+  } catch (...) {
+    jniUtil.CatchCppExceptionAndThrowJava(env);
+  }
 }
 
-JNIEXPORT jlong JNICALL Java_org_opensearch_knn_jni_NmslibService_loadIndex(JNIEnv * env, jclass cls,
-                                                                            jstring indexPathJ, jobject parametersJ)
-{
-    try {
-        return knn_jni::nmslib_wrapper::LoadIndex(&jniUtil, env, indexPathJ, parametersJ);
-    } catch (...) {
-        jniUtil.CatchCppExceptionAndThrowJava(env);
-    }
-    return NULL;
+JNIEXPORT jlong JNICALL Java_org_opensearch_knn_jni_NmslibService_loadIndex(JNIEnv *env, jclass cls,
+                                                                            jstring indexPathJ, jobject parametersJ) {
+  try {
+    return knn_jni::nmslib_wrapper::LoadIndex(&jniUtil, env, indexPathJ, parametersJ);
+  } catch (...) {
+    jniUtil.CatchCppExceptionAndThrowJava(env);
+  }
+  return NULL;
 }
 
-JNIEXPORT jobjectArray JNICALL Java_org_opensearch_knn_jni_NmslibService_queryIndex(JNIEnv * env, jclass cls,
+JNIEXPORT jlong JNICALL Java_org_opensearch_knn_jni_NmslibService_loadIndexWithStream(JNIEnv *env,
+                                                                                      jclass cls,
+                                                                                      jobject readStream,
+                                                                                      jobject parametersJ) {
+  try {
+    // Create a mediator locally.
+    // Note that `indexInput` is `IndexInputWithBuffer` type.
+    knn_jni::stream::NativeEngineIndexInputMediator mediator{&jniUtil, env, readStream};
```

Comment:
Do we need to create this here? Purpose for nmslib_wrapper separation was that it would be easier to uT if they werent included in the jni implementation

Reply:
yeah, we can move them to nmslib_wrapper and make JNI layer much thiner.
will update it.